### ISSUE TITLE
Additional layout utilities

### ISSUE THREAD

**Issue by lindsaym-fa** [shoelace-style/webawesome#165]
`wa-page` is a brilliant tool for full-page layouts, but how might we make it easier for devs to organize content within sections of a page?

We'll draw inspiration from https://every-layout.dev/, global design systems like Material, and front-end libraries like Bootstrap to see how others approach the problem and evaluate what devs might be familiar/comfortable with (and how we can make it better). Extra layout utilities should also benefit our own patterns, too.

**Comment by LeaVerou** [comment]
Great work! I spent some time looking at #169, and I’m recording some notes below that we can discuss sync _(I don't expect anyone to read these before the meeting, I need to jot things down otherwise I forget, and I figured if I’m writing them, I may as well share them…)_

## Overall / Big Picture

- **Goals**: Is the idea that these save time and effort from people that _could_ write the CSS if they had to, or are they also abstracting the knowledge required, i.e. enabling people to use these layouts that would otherwise _not_ be able to write the CSS for them? There's no right or wrong answer, but it’s important to be on the same page.
- I **love** the idea of using an existing recognized resource to draw inspiration for what types of layouts we need to support! One thing to be careful with is that the vast majority of our users will probably not be familiar with EL terms, so the terms we use need to be understandable in their own right. Some of these already are (e.g. stack, grid), but I found others a bit confusing (e.g. flank — I kept thinking of 🥩 😅).
- **Syntax**: I found it interesting that a lot of parameterization is done via class suffixes, almost like a more readable Tailwind. 
	- Intuitively it seems that some things are easier to specify in the HTML like that and others not. What’s the difference? Is it about repetition? One-offs are easier this way vs things that style multiple instances easier in the CSS? Or are there other factors at play too?
	- Strategically, since there is such a visceral reaction to Tailwindesque approaches by a large part of the community, I would suggest that each suffix should correspond to a CSS variable so that we can present it as a choice: _you can use this variable OR this suffix for efficiency if you prefer_. That would also facilitate aggregation, by allowing you to set this parameter for multiple elements at once.
- Thinking out loud, one of my personal big pain points with CSS layout is how flex requires too many wrappers and grid requires too much setup (a lot of which cannot be overridden on individual children). I wonder if there's a way to solve this here. E.g. would it make sense to have something like `<wa-page>` for sublayout *in addition* to these utilities?

## Docs & Demos

### Making examples more prominent

The **examples** are very powerful! I suggest making them more prominent. I found myself scrolling to the bottom of each page to figure out what each layout could be useful for. Especially some of these (e.g. split, frame) are very abstract without examples. Maybe put them at the top of the page, to set the stage (or smaller screenshots of them)? Or maybe even in the list too — collage of examples on one side, abstract icon on the other side.

### Reducing demo markup to separate from demo styling

**Demos:** as someone parachuting on this page, I was confused about what classes were functionally required for the layout to work and which were not. Would it be possible to restrict the markup to the bare minimum required for the layout to work and have hidden CSS for styling the demo? (I assume that's what all the `layout-example-*` classes are for?). 

E.g. instead of this code:

<img width="749" alt="image" src="https://github.com/user-attachments/assets/4e143a81-cc6d-49d1-b9ce-8b3e1b4f04f5">

could we have:

```html
<div class="wa-stack">
  <div class="wa-cluster:align-start" style="min-height: 8rem;">
    <div></div>
    <div></div>
    <div></div>
  </div>
  <div class="wa-cluster:align-end" style="min-height: 8rem;">
    <div></div>
    <div></div>
    <div></div>
  </div>
  <div class="wa-cluster:align-center" style="min-height: 8rem;">
    <div></div>
    <div></div>
    <div></div>
  </div>
  <div class="wa-cluster:align-stretch" style="min-height: 8rem;">
    <div></div>
    <div></div>
    <div></div>
  </div>
</div>
```

?
Does the demo component support a concept of "staging CSS"? If not, could it?

### Icons 

The icons in the main page are nicely stylized, but are a bit _too_ faded out — at first I thought they were loading something.
<img width="1197" alt="image" src="https://github.com/user-attachments/assets/ecfe5f07-ad49-48bc-9b18-d0a15c2e0a4e">

### Making the composition of demos more quickly understandable

I would love some way to at a glance see how a layout is composed of these utilities. E.g. here:

<img width="736" alt="image" src="https://github.com/user-attachments/assets/df76fef2-0ce8-4495-ada2-3cd5b6a335ae">

This actually uses three: stack, split, cluster, but without inspecting them with the dev tools, it took some time to figure out which one is which and how it all comes together. I wonder if we can address that in the demo component itself, e.g. you hover over an element and see it highlighted in the rendering.

## Notes on specific layouts

- Flank
	- I wonder if the mental model for these layouts is more commonly around marking the *element* that flanks (e.g. via a class), rather than the *direction* of flanking. 
- Frame
	- `:landscape` or `:portrait` are nicely readable, but I wonder if they communicate a more generic concept than their actual meaning (which is a specific aspect ratio). I wonder if something that reflects the actual value may be more understandable, e.g. `:16/9` (probably with a different character, as `/` would need escaping)
	- A shame we need to use `--aspect-ratio`. Can we not use `aspect-ratio`?
	- Modifiers like `:border-radius-2xl` teeter on the boundary of being too long to be useful as modifiers. Perhaps `:rounding-*`?
- Grid
	- A bit weird that `min-inline-size` refers to the element itself but `--min-inline-size` to the children, perhaps rename?
- Split
	- I’m a bit unclear on the difference with flank. Is it that a flank just has an item that doesn’t flex? But in the examples a split also seems to have items that don't flex?

## Nits

- Consistency: Some css variables are `wa-` prefixed and others are not, is this intentional?
- You may be aware, but some of the demos didn't work in iOS Safari
- I believe it should be _correspond **to**_, not _correspond **with**_ in this context.

**Comment by lindsaym-fa** [comment]
Thanks, @LeaVerou! I made a number of changes to #169 following your feedback here. Some items worth calling out:

- I limited the Tailwind-like class modifiers in favor of single-purpose utility classes (e.g., `wa-align-items-*`, `wa-border-radius-*`, and `wa-gap-*`)
- For **flank**: I didn't end up changing anything re: the mental model this round, but to be clear, I'm open to it! Naming has been a challenge for this class in general, and I couldn't figure out what to name the class for marking the flanking item that felt more natural than specifying its position on the parent element via `:end` or `:start`. Suggestions welcome; I just couldn't crack it.
- For **frame**: I'm hesitant to change `:landscape` and `:portrait` to be more literal. I like how they're recognizable for people that: a) don't care what the specific ratio is, just that the image is either wide or tall; and b) may struggle to remember what the numerators and denominators represent in an aspect ratio.
- For **split** vs. **flank**: 
  - Split uses `justify-content: space-between` to ensure that both items, no matter the size, are positioned at opposite ends of the container.
  - Flank uses just enough space for the 'flanked' item and allows the 'main' item to fill the rest of the container. 
  - The biggest difference to me is how the available space is intended to be filled — **split is often mostly whitespace** and **flank is always filled**.
